File: Objective-C/Algorhythm/Algorhythm/PlaylistDetailViewController.m
//
//  PlaylistDetailViewController.m
//  Algorhythm
//
//  Created by Taha Abbasi on 5/7/15.
//  Copyright (c) 2015 TahaAbbasi. All rights reserved.
//

#import "PlaylistDetailViewController.h"
#import "Playlist.h"

@interface PlaylistDetailViewController ()

@end

@implementation PlaylistDetailViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    if (self.playlist) {
        self.playlistCoverImage.image = self.playlist.playlistIconLarge;
        self.playlistCoverImage.backgroundColor = self.playlist.backgroundColor;
        self.playlistTitle.text = self.playlist.playlistTitle;
        self.playlistDescription.text = self.playlist.playlistDescription;
        
        self.playlistArtist0.text = self.playlist.playlistArtist[0];
        self.playlistArtist1.text = self.playlist.playlistArtist[1];
        self.playlistArtist2.text = self.playlist.playlistArtist[2];
    }
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}



@end

File: Swift/Algorythm/Algorythm/PlaylistDetailViewController.swift
//
//  PlaylistDetailViewController.swift
//  Algorythm
//
//  Created by Taha Abbasi on 5/30/15.
//  Copyright (c) 2015 TahaAbbasi. All rights reserved.
//

import UIKit

class PlaylistDetailViewController: UIViewController {


    var playlist: Playlist?
    
    @IBOutlet weak var playlistCoverImage: UIImageView!
    @IBOutlet weak var playlistTitle: UILabel!
    @IBOutlet weak var playlistDescription: UILabel!
    
    
    @IBOutlet weak var playlistArtist0: UILabel!
    @IBOutlet weak var playlistArtist1: UILabel!
    @IBOutlet weak var playlistArtist2: UILabel!
    @IBOutlet weak var playlistArtist3: UILabel!
    @IBOutlet weak var playlistArtist4: UILabel!
    @IBOutlet weak var playlistArtist5: UILabel!
    @IBOutlet weak var playlistArtist6: UILabel!
    
    override func viewDidLoad() {
        super.viewDidLoad()

        if playlist != nil {
            
            
            playlistCoverImage.image = playlist!.largeIcon
            playlistCoverImage.backgroundColor = playlist!.backgroundColor
            playlistTitle.text = playlist!.title
            playlistDescription.text = playlist!.description
            
            
            playlistArtist0.text = playlist!.artists[0]
            playlistArtist1.text = playlist!.artists[1]
            playlistArtist2.text = playlist!.artists[2]
            playlistArtist3.text = playlist!.artists[3]
            playlistArtist4.text = playlist!.artists[4]
            playlistArtist5.text = playlist!.artists[5]
            playlistArtist6.text = playlist!.artists[6]
            
        }
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
}
